extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -98,15 +98,7 @@ void registerBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans, Combi
                     ReactiveSessionFactoryProducer.class.getSimpleName());
             return;
         }
-        if (descriptors.size() == 1) {
-            // Only register the bean if their EMF dependency is also available, so use the same guard as the ORM extension
-            additionalBeans.produce(new AdditionalBeanBuildItem(ReactiveSessionFactoryProducer.class));
-        } else {
-            throw new ConfigurationException(
-                    "The Hibernate Reactive extension requires exactly one persistence unit configured: " + descriptors
-                            .stream()
-                            .map(PersistenceUnitDescriptorBuildItem::getPersistenceUnitName).collect(Collectors.toList()));
```

Comment:
Did you check we still give a meaningful error when trying to configure a named persistence unit as reactive?

Reply:
Not yet